frontend/src/components/stages/group_chat_participant_view.ts

```diff
@@ -212,6 +214,52 @@ export class GroupChatView extends MobxLitElement {
     `;
   }
 
+  private renderIndicators() {
+    if (!this.stage) return nothing;
+
+    const publicStageData = this.cohortService.stagePublicDataMap[
+      this.stage.id
+    ] as ChatStagePublicData;
+
+    // Check if timer has run out
+    const timeRemaining = getChatTimeRemainingInSeconds(
+      this.stage,
+      this.cohortService.chatMap,
+    );
+    if (timeRemaining !== null && timeRemaining <= 0) {
+      return html`
+        <div slot="indicators" class="description error">
+          <mwc-icon>timer_off</mwc-icon>
+          The time for this stage has ended.
```

Comment:
Note: Consider turning this into a "System" Message as per your latest PR so that agents can see this too? Make the consequences more clear too - "The timer for this stage has ended; you can no longer respond." (unless this is too long)